dotnet/src/Microsoft.Agents.AI.AGUI/AGUIAgent.cs

```diff
@@ -0,0 +1,102 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Runtime.CompilerServices;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Agents.AI.AGUI.Shared;
+using Microsoft.Extensions.AI;
+using Microsoft.Shared.Diagnostics;
+
+namespace Microsoft.Agents.AI.AGUI;
+
+/// <summary>
+/// Provides an <see cref="AIAgent"/> implementation that communicates with an AG-UI compliant server.
+/// </summary>
+public sealed class AGUIAgent : AIAgent
```

Comment:
I'm going to make this implement IChatClient instead in the next PR, as it will allow me to layer tool calling on top.